Re-wrote nessus parser to be more flexible

SferrellaA opened the pull request:
Re-wrote nessus parser to be more flexible

L1ghtn1ng commented:
Thanks for this.

Just 2 things.

First can you make this work with python3 as well as py2
Two can you use argparse please for arguments.

After that it is good to go.
________________________________
From: SferrellaA <notifications@github.com>
Sent: Friday, September 15, 2017 9:42:59 PM
To: leebaird/discover
Cc: Subscribed
Subject: [leebaird/discover] Re-wrote nessus parser to be more flexible (#107)


Re-wrote nessus parser to be more flexible

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/leebaird/discover/pull/107

Commit Summary

  *   Re-wrote nessus parser to be more flexible

File Changes

  *   A parsers/parse-nessus-old.py<https://github.com/leebaird/discover/pull/107/files#diff-0> (167)
  *   M parsers/parse-nessus.py<https://github.com/leebaird/discover/pull/107/files#diff-1> (213)

Patch Links:

  *   https://github.com/leebaird/discover/pull/107.patch
  *   https://github.com/leebaird/discover/pull/107.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/leebaird/discover/pull/107>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADddQoQseqaWnCUtOPKyW5jykZZHPkeJks5siuFTgaJpZM4PZfSB>.

SferrellaA commented:
I added python3 functionality to parse-nessus.py, but found that utfdictcsv.py wasn't python3 compatible either, so I updated both.

For argparse however, I'm not sure it's needed, as the only arguments that parse-nessus is called with are the names of files to parse

L1ghtn1ng commented:
The reason I say argparse is that it is neater in the code and means the script can be extended in the future easier.

Also can you please specify the actual error exception that gets generated by utfdictcsv.py when it errors out, instead of having a try and except block that catches all errors please. This is one on the pep8 rules.

Nearly there at getting it merged ☺️

Sent from phone



From: SferrellaA
Sent: Thursday, 21 September, 19:13
Subject: Re: [leebaird/discover] Re-wrote nessus parser to be more flexible (#107)
To: leebaird/discover
Cc: J.Townsend, Comment


I added python3 functionality to parse-nessus.py, but found that utfdictcsv.py wasn't python3 compatible either, so I updated both.
For argparse however, I'm not sure it's needed, as the only arguments that parse-nessus is called with are the names of files to parse
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/leebaird/discover/pull/107#issuecomment-331238301>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADddQj8m6dX6UXHfqT8_RbSlVy-RIwM9ks5skqdMgaJpZM4PZfSB>.
